pajbot/web/routes/base/login.py

```diff
@@ -166,3 +195,67 @@ def logout():
     @twitch.tokengetter
     def get_twitch_oauth_token():
         return session.get("twitch_token")
+
+    if not spotify:
+        return
+
+    @app.route("/spotify_login")
+    def spotify_login():
+        if spotify is not None:
+            callback_url = (
+                app.bot_config["spotify"]["redirect_uri"]
+                if "redirect_uri" in app.bot_config["spotify"]
+                else url_for("authorized", _external=True)
+            )
+            state = request.args.get("n") or request.referrer or None
+            return spotify.authorize(
+                callback=callback_url, state=state, scope=" ".join(spotify_scopes), force_verify="true"
+            )
+        return render_template("login_error.html")
+
+    @app.route("/login/spotify_auth")
+    def spotify_auth():
+        try:
+            resp = spotify.authorized_response(spotify=True)
+        except OAuthException as e:
+            log.error(e)
+            log.exception("An exception was caught while authorizing")
+            next_url = get_next_url(request, "state")
+            return redirect(next_url)
+        except Exception as e:
+            log.error(e)
+            log.exception("Unhandled exception while authorizing")
+            return render_template("login_error.html")
+
+        session["spotify_token"] = (resp["access_token"],)
```

Comment:
what is the point of storing the spotify token on the session?

Reply:
The oauth token getter uses it